src/main/java/net/greatstart/controllers/PasswordResetController.java

```diff
@@ -0,0 +1,84 @@
+package net.greatstart.controllers;
+
+import net.greatstart.model.GenericResponse;
+import net.greatstart.model.User;
+import net.greatstart.services.SecurityService;
+import net.greatstart.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.core.env.Environment;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Locale;
+import java.util.UUID;
+
+@Controller
+public class PasswordResetController {
+
+    private MessageSource messages;
+    private Environment env;
+    private JavaMailSender mailSender;
+    private UserService userService;
+    private SecurityService securityService;
+
+    @Autowired
+    public PasswordResetController(MessageSource messages, Environment env, JavaMailSender mailSender, UserService
+        userService, SecurityService securityService) {
+        this.messages = messages;
+        this.env = env;
+        this.mailSender = mailSender;
+        this.userService = userService;
+        this.securityService = securityService;
+    }
+
+    @PostMapping(value = "/user/resetPassword")
+    @ResponseBody
+    public GenericResponse resetPassword(HttpServletRequest request, @RequestParam("email") String userEmail) {
+        User user = userService.getUserByEmail(userEmail);
+        if (user == null) {
+            throw new RuntimeException("Email not found");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)